core-bundle/src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -2778,6 +2796,20 @@ protected function generateTree($path, $intMargin, $mount=false, $blnProtected=t
 
 			$strFileNameEncoded = StringUtil::convertEncoding(StringUtil::specialchars(basename($currentFile)), Config::get('characterSet'));
 
+            // Call the label_callback ($strFileNameEncoded, $thumbnail, $objFile, $protected, $this)
+            if (\is_array($GLOBALS['TL_DCA'][$this->strTable]['list']['label']['label_callback']))
+            {
+                $strClass = $GLOBALS['TL_DCA'][$this->strTable]['list']['label']['label_callback'][0];
+                $strMethod = $GLOBALS['TL_DCA'][$this->strTable]['list']['label']['label_callback'][1];
+
+                $this->import($strClass);
+                list($strFileNameEncoded, $thumbnail) = $this->$strClass->$strMethod($strFileNameEncoded, $thumbnail, $objFile, $protected, $this);
```

Comment:
I would prefer not to return anything here and modify the values via reference instead.